hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java

```diff
@@ -467,11 +599,89 @@ static Builder portable(@Nonnull ClassDefinition classDefinition) {
         Builder writeGenericRecord(@Nonnull String fieldName, @Nullable GenericRecord value);
 
         /**
+         * Writes a decimal which is arbitrary precision and scale floating-point number
          * It is illegal to write to the same field twice.
          *
          * @param fieldName name of the field as it is defined in its class definition.
          *                  See {@link ClassDefinition} for {@link Portable}
-         * @param value
+         * @param value     to set to GenericRecord
+         * @return itself for chaining
+         * @throws HazelcastSerializationException if the field name does not exist in the class definition or
+         *                                         the type of the field does not match the one in the class definition or
+         *                                         Same field is trying to be overwritten without using
+         *                                         {@link GenericRecord#cloneWithBuilder()}.
+         */
+        Builder writeDecimal(@Nonnull String fieldName, @Nullable BigDecimal value);
```

Comment:
nit: might be a good idea to mark return type with `@Nonnull` here and in other builder methods

Reply:
Somehow I find it awkward to put @Nonnull to builder pattern. Probably because it is obvious but, I will put it for the sake of completeness.

Reply:
IDE and other tools like Kotlin compiler can't recognize builder pattern while these annotations are valuable for them.